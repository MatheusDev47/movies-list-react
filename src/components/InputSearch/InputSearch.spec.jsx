import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { InputSearch } from '.';

describe('<InputSearch', () => {
  it('Should have a value in searchValue', () => {
    const fn = jest.fn();

    render(<InputSearch onChange={fn} value="Test Value" />);

    const input = screen.getByPlaceholderText(/search for movies/i);

    expect(input.value).toBe('Test Value');
  });

  it('Should call onChange function when key is press', () => {
    const fn = jest.fn();

    render(<InputSearch onChange={fn} value="test onChange function" />);

    const input = screen.getByPlaceholderText(/search for movies/i);
    const valueInputTest = 'value test';

    userEvent.type(input, valueInputTest);

    expect(input.value).toBe('test onChange function');
    expect(fn).toHaveBeenCalledTimes(valueInputTest.length);
  });

  it('Should match snapshot', () => {
    const fn = jest.fn();

    const { container } = render(<InputSearch onChange={fn} value="" />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
