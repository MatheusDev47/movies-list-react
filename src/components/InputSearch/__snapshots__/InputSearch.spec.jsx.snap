// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputSearch Should match snapshot 1`] = `
<input
  placeholder="Search for movies"
  type="search"
  value=""
/>
`;
